influxql/select.go

```diff
@@ -132,6 +132,9 @@ func buildFieldIterators(fields Fields, ic IteratorCreator, opt IteratorOptions)
 			// Build iterators for calls first and save the iterator.
 			// We do this so we can keep the ordering provided by the user, but
 			// still build the Call's iterator first.
+
+			f.Expr = Reduce(f.Expr, nil)
```

Comment:
Can you assign the reduced expression to a temporary variable? I would prefer not to modify the array that is coming into this function to avoid unintended side effects at the call site. You can do the reduction here after checking for a variable reference and then do it one more time below when the auxiliary iterators are being created.